OmsAgent/omsagent.py

```diff
@@ -217,7 +218,15 @@ def telemetry():
         f.write(str(py_pid) + '\n')
 
     watcher = watcherutil.Watcher(HUtilObject.error, HUtilObject.log, log_to_console=True)
-    watcher.watch()
+
+    watcher_thread = Thread(target = watcher.watch)
+    self_mon_thread = Thread(target = watcher.monitor_health)
```

Comment:
Why don't we do everything on the watcher thread? The health monitoring methods can be called from there too.

Reply:
In my opinion telemetry and self monitoring threads have different responsibility and should be separate. One should not block other. Actualy i am even thinking of taking self monitoring our of watcherutil into separate python script , but will do that later. That will help me to put it in the bundle instead of extension.

Reply:
I was thinking along the lines of that every periodic activity carried out by the extension should be plugged into a unified framework. In any case, if you are satisfied with this approach, that's fine.